src/libraries/Common/src/System/Security/Cryptography/SlhDsa.cs

```diff
@@ -726,26 +971,36 @@ public static SlhDsa ImportSubjectPublicKeyInfo(ReadOnlySpan<byte> source)
                         AsnValueReader reader = new AsnValueReader(source, AsnEncodingRules.DER);
                         SubjectPublicKeyInfoAsn.Decode(ref reader, manager.Memory, out SubjectPublicKeyInfoAsn spki);
 
-                        SlhDsaAlgorithm algorithm = SlhDsaAlgorithm.GetAlgorithmFromOid(spki.Algorithm.Algorithm);
+                        SlhDsaAlgorithm algorithm = GetAlgorithmIdentifier(ref spki.Algorithm);
 
-                        if (spki.Algorithm.Parameters.HasValue)
+                        try
                         {
-                            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
-                            spki.Algorithm.Encode(writer);
-                            throw Helpers.CreateAlgorithmUnknownException(writer);
+                            return ImportSlhDsaPublicKey(algorithm, spki.SubjectPublicKey.Span);
+                        }
+                        catch (ArgumentException ae)
+                        {
+                            throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding, ae);
                         }
-
-                        return SlhDsaImplementation.ImportPublicKey(algorithm, spki.SubjectPublicKey.Span);
                     }
                 }
             }
         }
 
+        /// <inheritdoc cref="ImportSubjectPublicKeyInfo(ReadOnlySpan{byte})" />
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        public static SlhDsa ImportSubjectPublicKeyInfo(byte[] source)
+        {
+            ArgumentNullException.ThrowIfNull(source);
+            return ImportSubjectPublicKeyInfo(source.AsSpan());
```

Comment:
```suggestion
            return ImportSubjectPublicKeyInfo(new ReadOnlySpan<byte>(source));
```

AsSpan makes a writable span, which then calls the implicit conversion operator.  Sure, maybe the JIT inlines it, but let's just get there in one transformation.